test/e2e/deployment-plans/interior.go

```diff
@@ -127,6 +136,43 @@ func testInteriorScaleUp(f *framework.Framework) {
 	By("Verifying the Deployment has reached 4 replicas")
 	err = framework.WaitForDeployment(f.KubeClient, f.Namespace, "interior-interconnect", 4, framework.RetryInterval, framework.Timeout)
 	Expect(err).NotTo(HaveOccurred())
+
+	By("Verifying the Deployment contains 4 pods")
+	pods, err := f.ListPodsForDeployment(dep)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(len(pods.Items)).To(Equal(4))
+
+	By("Verifying the Network contains 4 nodes on each pod")
+	for _, pod := range pods.Items {
+		var nodes []entities.Node
+		// Retry logic to retrieve nodes
+		err = wait.Poll(5*time.Second, 20*time.Second, func() (done bool, err error) {
+			if nodes, err = router_mgmt.QdmanageQueryNodes(f, pod.Name); err != nil {
+				return false, err
+			}
+			if len(nodes) != 4 {
+				return false, nil
+			}
+			return true, nil
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(nodes)).To(Equal(4))
+	}
+
+	By("Verifying each node has 3 inter-router connections")
```

Comment:
Similar to above, introduce `f.ListConnectionsForPod(pod)`...

I can imagine we will have a bunch of f.ListXYZForPod that are connection, links, addressess, etc.